orchestrator/graphql/utils/get_subscription_product_blocks.py

```diff
@@ -79,3 +81,24 @@ def included(key: str, value: Any) -> bool:
 
     product_blocks = (to_product_block(product_block) for product_block in get_all_product_blocks(subscription, tags))
     return [product_block for product_block in product_blocks if product_block.product_block_instance_values]
+
+
+async def get_subscription_product_blocks_json_schema(subscription_id: UUID) -> dict:
+    subscription_model = SubscriptionModel.from_subscription(subscription_id)
+    subscription = build_extended_domain_model(subscription_model)
+    block_paths = product_block_paths(subscription)
+
+    root_block_paths = [path for path in block_paths if "." not in path and path != "product"]
```

Comment:
This is a whole lot of work (database queries to load the subscription + domain model serialization) just to get the schema.

There should not be a reason to serialize the subscription itself; it would suggest that for example 2 firewall subscriptions can have a different JSON Schema. But they will always have the same schema. Deriving the schema off of the product type would be preferrable.

I'll admit I don't know straight away if/how that is even possible. If not, we could consider adding the schema to the domain-model cache

Reply:
We can query the subscription's product name, get the product type from the registry and call `.schema()` on it;

```
from orchestrator.domain import SUBSCRIPTION_MODEL_REGISTRY


SUBSCRIPTION_MODEL_REGISTRY['SN8 Corelink'].schema()
```

This schema seems to contain everything you need

Reply:
one issue with this, is that the SUBSCRIPTION_MODEL_REGISTRY has the subscriptions with active lifecycle, so we can't trust in the required fields, since some values can still be none. is that acceptable?

Reply:
I agree with mark, it should be possible to just get the right schema, by looking at the product type column, no need to serialise the whole model.

Reply:
oke, then the updated behaviour is fine, gets the product from db and try to get `SUBSCRIPTION_MODEL_REGISTRY` product class and take the base type to get the schema for it